File: Faza6/OnlineGamesShop/src/main/java/com/rich/onlinegamesshop/model/Publisher.java
package com.rich.onlinegamesshop.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Entity
@NoArgsConstructor
public class Publisher {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long publisherId;

    @Column(nullable = false)
    String name;

    @Column(nullable = false)
    String website;

    @Column(nullable = false)
    String country;

    public Publisher(String name, String website, String country) {
        this.name = name;
        this.website = website;
        this.country = country;
    }
}

File: Faza6/FrontEnd/Angular/src/app/model/Publisher.ts

export class Publisher{
  id_publisher:number;
    name:string;
    website:string;
    country:string;
    constructor(id_publisher:number,name:string,
                website:string,country:string){
        this.id_publisher=id_publisher;
        this.name=name;
        this.website=website;
        this.country=country;
    }
}
